OpenRA.Mods.Common/Traits/AppearsOnRadar.cs

```diff
@@ -40,7 +40,31 @@ public void Created(Actor self)
 
 		public IEnumerable<Pair<CPos, Color>> RadarSignatureCells(Actor self)
 		{
-			var color = modifier != null ? modifier.RadarColorOverride(self) : self.Owner.Color.RGB;
+			var player = self.World.RenderPlayer ?? self.World.LocalPlayer;
+			var color = self.Owner.Color.RGB;
+
+			if (modifier != null)
+				color = modifier.RadarColorOverride(self);
+
+			if (Game.Settings.Game.AllianceDisplay && player != null && !player.Spectating)
```

Comment:
Can you define a helper method instead of duplicating this code between here and the health bars?